Api/Connection.php

```diff
@@ -21,6 +20,14 @@ public function post($url, $bodyArray = false)
                 'Connection: Close',
                 'Host: '.$parsedUrl['host']
             ];
+            
+            if ($hmacAuth) {
+                $secret = $bodyArray['secret'];
+                unset($bodyArray['secret']);
+                $headers[] = 'HTTP_X_DIGEST: ' . hash_hmac('sha256', json_encode($bodyArray), $secret);
+            }
+
+            $encodedBody = $bodyArray ? json_encode($bodyArray) : '';
```

Comment:
what's the point in json encoding it two times instead of once?

Reply:
Since i don't have direct access to the api secret in Connection class i am passing it as part of the $data array in the Activity Helper. The double json encoding is happening because of the way Im passing secret information and the limitation of the PHP that cannot work with json formatted arrays without fist decoding them to standard PHP arrays.